iron-form.html

```diff
@@ -286,6 +286,7 @@
             node.value = defaults.value;
             node.checked = defaults.checked;
           }
+          node.invalid = false;
```

Comment:
I don't think this is correct -- what if the node is, in fact, invalid? (for example it's a `paper-input` with `required` set)

Reply:
Thank you for your response! Interesting point... However according to convention that is used by paper & iron elements, `invalid` property reflects state of the UI (visibility of error message), not the real validity. E.g. empty `paper-input` with `required` and `auto-validate` attributes has got `invalid == false` initially. It gets set to `invalid=true` only after validate() is called or user starts typing. For this reason I think there is no issue with setting invalid to false - we really want to reset the UI state.

Additionally, I don't think there is any other way of hiding the error messages...

Reply:
Well since the values are reset on every element, the element itself should react to that value changing, and clear the validation itself, no?

Reply:
Hmm... Let's consider `paper-input`. If I understand properly, you suggest to clear validation (hide red underline & error message) on every external change of its `.value` property. I think this would be a problem for React-style programming - people may want to update `.value` from above by themselves after each keystroke. In this case validation reset would be undesirable.

Just for inspiration - in __Angular__ there are actually two+ properties/flags for each form control - `invalid` that really describes whether model data is valid, and `dirty` that flags that data validity should be communicated in the GUI (whether user already tried to type something). Adding `dirty` reflected property into all Polymer paper form controls might be a one solution...

On the other hand in plain __React__ there is no dirty flag. In [Material UI](https://material-ui-next.com/demos/text-fields/) - popular React GUI component library - TextField doesn't care about validation and 'dirtiness' at all, its GUI state/appearance is a direct function of it's properties (show/don't show error message etc.), any validation is up to user.

As I said and as mentioned in https://github.com/PolymerElements/paper-input/issues/238, right now `invalid` property doesn't really reflect the real validity, but rather error message/red underline visibility. I don't find it to be an issue, even though it might be a bit confusing for starting programmers.